crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,248 @@
+        let aria_roles = ctx.aria_roles();
+        let attributes = ctx.extract_attributes(&node.attributes());
+        let element_name = element_name.text_trimmed();
+
+        if let Some(attributes_ref) = attributes.as_ref() {
+            let has_interactive_handler = CATEGORIES_TO_CHECK.iter().any(|&category| {
+                if let Some(handlers) = EVENT_TO_HANDLERS.get(category) {
+                    handlers.iter().any(|&handler| {
+                        if let Some(values) = attributes_ref.get(handler) {
+                            values.iter().any(|value| value != "null")
+                        } else {
+                            false
+                        }
+                    })
+                } else {
+                    false
+                }
+            });
+
+            if !has_interactive_handler {
+                return None;
+            }
+        } else {
+            return None;
+        }
+
+        if node
+            .find_attribute_by_name("aria-hidden")
+            .map_or(false, |attr| {
+                attr.as_static_value()
+                    .map_or(true, |val| val.text() == "true")
+            })
+        {
+            return None;
+        }
+
+        let is_valid_element = match element_name {
+            "section" => ["aria-label", "aria-labelledby"].iter().any(|&attr_name| {
+                node.find_attribute_by_name(attr_name)
+                    .map_or(false, |attr| {
+                        attr.as_static_value()
+                            .map_or(false, |val| !val.text().is_empty())
+                    })
+            }),
+            "a" => node.find_attribute_by_name("href").map_or(false, |attr| {
+                attr.as_static_value()
+                    .map_or(false, |val| !val.text().is_empty())
+            }),
+            _ => {
+                (!aria_roles.is_not_interactive_element(element_name, attributes.clone())
+                    && !is_invalid_element(element_name))
+                    || is_valid_element(element_name)
+            }
+        };
+
+        if is_valid_element {
+            return None;
+        }
+
+        if let Some(attr) = node.find_attribute_by_name("role") {
+            let role_value = attr.as_static_value()?;
+            let role_text = role_value.text();
+
+            if aria_roles.is_role_interactive(role_text) {
+                return None;
+            }
+        } else {
+            return Some(());
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {{"Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element"}},
```

Comment:
```suggestion
            markup! {{"Avoid non-native interactive elements."}},
```

Reply:
@unvalley 
fixed in this https://github.com/biomejs/biome/pull/2981/commits/53745f3d518a3ef7739c01afc67bd7a126e37432